src/API/Trace/Propagation/TraceContextPropagator.php

```diff
@@ -97,15 +97,20 @@ private static function extractImpl($carrier, PropagationGetterInterface $getter
         // traceParent = {version}-{trace-id}-{parent-id}-{trace-flags}
         $pieces = explode('-', $traceparent);
 
+        // If the header does not have at least 4 pieces, it is invalid -- restart the trace.
+        if (count($pieces) < 4) {
+            return SpanContext::getInvalid();
+        }
+
         [$version, $traceId, $spanId, $traceFlags] = $pieces;
 
         $versionIsFuture = hexdec($version) > hexdec(self::VERSION);
-        $versionIsInvalid = !preg_match(self::HEX_REGEX, $version)
+        $versionIsInvalid = !preg_match(self::VERSION_REGEX, $version)
```

Comment:
```php
hexdec($version) === 0xff
```

Reply:
I believe the version regex `/^(?!ff)[\da-f]{2}$/` will take care of both `ff` and non-hex or >2 char hex sequences.

Reply:
Non-hex has be checked before calling `hexdec()` in the line above this, otherwise it might trigger a deprecation notice.

Reply:
Yes, absolutely! This was triggering an error on tests on I'm in the process of writing. Let me get this in order.